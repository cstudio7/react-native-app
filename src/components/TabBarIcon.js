import React from 'react';
import { Colors } from '../constants';
import Icon from 'react-native-vector-icons/Ionicons';

const TabBarIcon = ({ name, focused }) => (
  <Icon
    name={name}
    size={26}
    style={{ marginBottom: -5 }}
    color={focused ? Colors.primary : Colors.gray}
  />
);

export default TabBarIcon;
