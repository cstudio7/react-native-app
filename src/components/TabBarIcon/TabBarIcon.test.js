import React from 'react';
import renderer from 'react-test-renderer';
import TabBarIcon from './TabBarIcon';

it('renders heart icon', () => {
  const tree = renderer.create(<TabBarIcon name="md-heart" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders focused heart icon', () => {
  const tree = renderer
    .create(<TabBarIcon name="md-heart" focused={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders list icon', () => {
  const tree = renderer.create(<TabBarIcon name="md-list" />).toJSON();
  expect(tree).toMatchSnapshot();
});
