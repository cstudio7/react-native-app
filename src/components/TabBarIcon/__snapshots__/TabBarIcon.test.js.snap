// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders focused heart icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      Array [
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#673AB7",
          "marginBottom": -5,
        },
      ],
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders heart icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      Array [
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#9e9e9e",
          "marginBottom": -5,
        },
      ],
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders list icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      Array [
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#000",
          "fontSize": 30,
        },
        Object {
          "color": "#9e9e9e",
          "marginBottom": -5,
        },
      ],
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
